src/jep/convert_p2j.c

```diff
@@ -52,6 +56,58 @@ static jmethodID unmodifiableList      = 0;
     static jstring UTF8 = NULL;
 #endif
 
+bool isFunctionalInterfaceType(JNIEnv *env, jclass type) {
+    if (!JNI_METHOD(classGetMethods, env, JCLASS_TYPE, "getMethods", "()[Ljava/lang/reflect/Method;")) {
+        process_java_exception(env);
+        return NULL;
+    }
+    if (!JNI_METHOD(classGetModifiers, env, JCLASS_TYPE, "getModifiers", "()I")) {
+        process_java_exception(env);
+        return NULL;
+    }
+    if (!JNI_METHOD(methodGetModifiers, env, JMETHOD_TYPE, "getModifiers", "()I")) {
+        process_java_exception(env);
+        return NULL;
+    }
+    if ((*env)->PushLocalFrame(env, JLOCAL_REFS) != 0) {
+        process_java_exception(env);
+        return NULL;
+    }
+    int classModifiers = (*env)->CallIntMethod(env, type, classGetModifiers);
+    if (!(classModifiers & 512)) {
+        return NULL; // It's not an interface, so it can't be functional
+    }
+    jobjectArray methods = (jobjectArray) (*env)->CallObjectMethod(env, type, classGetMethods);
+    if (process_java_exception(env)) {
+        (*env)->PopLocalFrame(env, NULL);
+        return NULL;
+    }
+    jsize numMethods = (*env)->GetArrayLength(env, methods);
+    jobject abstractMethod = NULL;
+    for (jsize i = 0; i < numMethods; i++) {
+        jobject method = (*env)->GetObjectArrayElement(env, methods, i);
+        jint modifiers = (*env)->CallIntMethod(env, method, methodGetModifiers);
+        if (process_java_exception(env)) {
+            (*env)->PopLocalFrame(env, NULL);
+            return NULL;
+        }
+        if (modifiers & 1024) { // check if the method is abstract
+            // We found two different abstract methods, so we're not a functional interfaces
+            if (abstractMethod != NULL) {
+                (*env)->PopLocalFrame(env, NULL);
+                return NULL;
+            } else {
+                abstractMethod = method;
+            }
```

Comment:
Comment of "We found two different abstract methods..." should go inside `if (abstractMethod != NULL)`.  I was confused first time reading it.